src/component/project/ProjectAccess/ProjectAccessAssign/ProjectAccessAssign.tsx

```diff
@@ -169,15 +149,28 @@ export const ProjectAccessAssign = ({
         [users, accesses, edit, groups]
     );
 
-    useEffect(() => {
-        const selectedOption =
+    const [selectedOptions, setSelectedOptions] = useState<IAccessOption[]>(
+        () =>
             options.filter(
                 ({ id, type }) =>
                     id === selected?.entity.id && type === selected?.type
-            ) || [];
-        setSelectedOptions(selectedOption);
-        setRole(roles.find(({ id }) => id === selected?.entity.roleId) || null);
-    }, [open, selected, options, roles]);
+            )
+    );
+    const [role, setRole] = useState<IProjectRole | null>(
+        roles.find(({ id }) => id === selected?.entity.roleId) ?? null
+    );
+
+    const payload = useMemo(
```

Comment:
I would like to do some research on the inflection point where useMemo makes sense. It's not worth complicating the code if it does not give us a tangible benefit. I suspect we'll be surprised at where the inflection point is.